packages/perseus/src/validation.types.ts

```diff
@@ -236,58 +238,105 @@ export type PerseusTableRubric = {
 
 export type PerseusTableUserInput = ReadonlyArray<ReadonlyArray<string>>;
 
-export type Rubric =
-    | PerseusCategorizerScoringData
-    | PerseusCSProgramRubric
-    | PerseusDropdownRubric
-    | PerseusExpressionRubric
-    | PerseusGroupRubric
-    | PerseusGradedGroupRubric
-    | PerseusGradedGroupSetRubric
-    | PerseusGrapherRubric
-    | PerseusIFrameRubric
-    | PerseusInputNumberRubric
-    | PerseusInteractiveGraphRubric
-    | PerseusLabelImageRubric
-    | PerseusMatcherRubric
-    | PerseusMatrixRubric
-    | PerseusNumberLineScoringData
-    | PerseusNumericInputRubric
-    | PerseusOrdererRubric
-    | PerseusPlotterScoringData
-    | PerseusRadioRubric
-    | PerseusSorterRubric
-    | PerseusTableRubric;
-
-export type UserInput =
-    | PerseusCategorizerUserInput
-    | PerseusCSProgramUserInput
-    | PerseusDropdownUserInput
-    | PerseusExpressionUserInput
-    | PerseusGrapherUserInput
-    | PerseusGroupUserInput
-    | PerseusIFrameUserInput
-    | PerseusInputNumberUserInput
-    | PerseusInteractiveGraphUserInput
-    | PerseusLabelImageUserInput
-    | PerseusMatcherUserInput
-    | PerseusMatrixUserInput
-    | PerseusNumberLineUserInput
-    | PerseusNumericInputUserInput
-    | PerseusOrdererUserInput
-    | PerseusPlotterUserInput
-    | PerseusRadioUserInput
-    | PerseusSorterUserInput
-    | PerseusTableUserInput;
-
-export type UserInputMap = {[widgetId: string]: UserInput};
+export interface RubricRegistry {
+    categorizer: PerseusCategorizerScoringData;
+    "cs-program": PerseusCSProgramRubric;
+    // definition: PerseusDefinitionScoringData;
```

Comment:
I really am not a fan of how this works right now. The fact that this is not a full listing of all widgets because we don't score all widgets worries me, but I don't think it makes sense to add a type for all widgets just to make this "whole". 

I'm tempted to leave it like this for now and as we work further hopefully something resolves to a better solution. 

Also open to ideas.

Reply:
If not all widgets have ScoringData, why don't we just remove the inner comments and add a comment to the whole interface (just above line 241) saying that? We could keep a list of which widgets currently don't have scoring data in that comment so it doesn't muddy the interface so to speak. Or is there a reason why we _should_ have an entry for every widget in this interface?